lib/improver/utilities/cube_extraction.py

```diff
@@ -78,6 +79,33 @@ def is_complex_parsing_required(value):
     return complex_constraint
 
 
+def create_constraint(value):
+    """
+    Constructs an appropriate contraint for matching numerical values is they
+    are floating point. If not, the original values are returned as a list
+    (even if they were single valued on entry).
+
+    Args:
+        value (float/int or list of float/int):
+            Contraint values that are being used to match against values in a
+            cube for the purposes of extracting elements of the cube.
+    Returns:
+        lambda function or list:
+            If the input value(s) are floating point this function returns a
+            lambda function that will enable for approximate matching to
+            ensure they can be matched to cube values. If the inputs are int
+            or non-numeric, they will be returned unchanged, except for single
+            values that will have become lists.
+    """
+    if not isinstance(value, list):
+        value = [value]
+
+    if np.issubdtype(np.array(value).dtype, np.floating):
```

Comment:
Potentially, it might be best to just return the return lambda cell: any(np.isclose(cell.point, value)), regardless whether they are floating point. The rationale for this is linked to the unit conversion within the `apply_extraction` function, as I could potentially input a cube with an `air_temperature` coordinate in Kelvin but define the values to extract in Celsius as integers. The unit conversion, at the moment, can turn a list of integer into a list of floats, for example 283.15 K will be converted into 10.000001, rather than 10, meaning that a constraint aiming to extract the 10C threshold, won't find any matches. What do you think?

Reply:
Agreed. Changed, whilst preserving strings as strings.